### PR TITLE
Auth flow

### PR DESCRIPTION
**Pull request by bluquar**
Support these flows:
1. Create account -> verify account -> home -> profile -> logout
2. Create account -> unable to verify -> logout -> create account
3. Login -> home -> profile -> logout

Not yet supported:
1. Create account -> unable to verify -> resend email
2. Login -> forgot password

https://user-images.githubusercontent.com/4309735/143394037-7d52f06e-8d40-4c93-b323-1732cc9baf20.mov

### CODE REVIEW

File: screens/CreateAccountScreen.tsx [hunk 1]
```diff
@@ -16,7 +16,9 @@ import type {ApolloError, MutationResult} from '@apollo/client';
 
 WebBrowser.maybeCompleteAuthSession();
 
-export default function RegisterScreen(_: RootTabScreenProps<'Register'>) {
+export default function CreateAccountScreen(
+  _: RootStackScreenProps<'Create Account'>,
```
Comment (lowellbander) [inline screens/CreateAccountScreen.tsx:20]: why underscore here rather than like screenProps or sth?

File: screens/HomeScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+import {Text, View} from '../components/Themed';
+import {RootTabScreenProps} from '../types';
+import {StyleSheet} from 'react-native';
+
+export default function HomeScreen({
+  navigation: _,
```
Comment (lowellbander) [inline screens/HomeScreen.tsx:8]: is the underscore the type here? how's that work?

File: screens/LoginScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+`;
+
+function processLoginMutationState({
+  data,
+  loading,
+  error,
+}: MutationResult<TMutationData>): TProcessedMutationData {
+  return {
+    authToken: data?.tokenAuth?.token,
+    didSuccessfullyLogin: data?.tokenAuth?.success || false,
```
Comment (lowellbander) [inline screens/LoginScreen.tsx:103]: any reason u prefer this to `Boolean(val)`?

File: utils/PersistentClientStorage/index.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+import client from '../../ca_graphql/client';
+
+type Callback = (value: null | string) => void;
```
Comment (lowellbander) [inline utils/PersistentClientStorage/index.ts:6]: does typescript not permit you to write `null | string` as `?string`?

File: utils/PersistentClientStorage/index.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+class PersistentClientStorage {
+  data: Map<string, string | null> = new Map();
```
Comment (lowellbander) [inline utils/PersistentClientStorage/index.ts:9]: ooo why this as opposed to Object?

File: utils/PersistentClientStorage/index.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  loading = true;
+  subscribers: Map<string, Set<Callback>> = new Map();
```
Comment (lowellbander) [inline utils/PersistentClientStorage/index.ts:11]: Soybeannnnnnnnnnnnnn

**Review by lowellbander — APPROVED** [review]
amazing fuckin work bud always cool to see you write these fun custom modules :)

File: screens/CreateAccountScreen.tsx (hunk 1 above)
Comment (bluquar) [reply screens/CreateAccountScreen.tsx:20]: I set up eslint so unused variables need to start with an underscore. This just indicates that it's unused, but I'm specifying the type to make it easier if we want to use the props later. I could call it "_props" tho

File: screens/HomeScreen.tsx (hunk 2 above)
Comment (bluquar) [reply screens/HomeScreen.tsx:8]: This is syntax for object destructuring and renaming. It's equivalent to "const _ = props.navigation"

File: screens/LoginScreen.tsx (hunk 3 above)
Comment (bluquar) [reply screens/LoginScreen.tsx:103]: Nope!

File: utils/PersistentClientStorage/index.ts (hunk 4 above)
Comment (bluquar) [reply utils/PersistentClientStorage/index.ts:6]: It does not :(

File: utils/PersistentClientStorage/index.ts (hunk 5 above)
Comment (bluquar) [reply utils/PersistentClientStorage/index.ts:9]: I prefer the syntax and semantics for Map. I'm gonna start using Object only for "shapes", not for things with homogeneous key/val types. Can elaborate more upon request

File: utils/PersistentClientStorage/index.ts (hunk 6 above)
Comment (bluquar) [reply utils/PersistentClientStorage/index.ts:11]: Lmao yeah I was worried Fb would think I'm stealing code from them

**Pull request merged**